app/src/main/java/com/xmartlabs/scasas/doapp/ui/grouplist/GroupsListFragment.java

```diff
@@ -0,0 +1,397 @@
+        .findFirst()
+        .get();
+  }
+
+  private void insertTask(Task task, Group group) {
+    taskController.insertTask(task)
+        .subscribe(new SingleSubscriber<Task>() {
+          @Override
+          public void onSuccess(Task task) {
+            Toast.makeText(getContext(), R.string.task_added, Toast.LENGTH_SHORT).show();
+            updateTaskCount(group);
+            closeNewTaskView();
+          }
+
+          @Override
+          public void onError(Throwable error) {
+            Timber.e(error);
+          }
+        });
+  }
+
+  private void updateAll(List<Group> groups) {
+    Stream.of(groups).forEach(this::updateTaskCount);
+  }
+
+  private void updateTaskCount(@NonNull Group group) {
+    taskController.getTasksCount(user, group).subscribe(new SingleSubscriber<Long>() {
+      @Override
+      public void onSuccess(Long taskCount) {
+        updateItemField(group.getName(), taskCount);
+      }
+
+      @Override
+      public void onError(Throwable error) {
+        Timber.e(error);
+      }
+    });
+  }
+
+  private void updateItemField(String groupName, Long itemCount) {
+    if (Objects.equals(groupName, GroupEnum.Shop.toString())) {
+      shopItemsView.setText(String.format(getString(R.string.item), itemCount));
+      return;
+    }
+    if (Objects.equals(groupName, GroupEnum.Work.toString())) {
+      shopItemsView.setText(String.format(getString(R.string.item), itemCount));
+      return;
+    }
+    if (Objects.equals(groupName, GroupEnum.Health.toString())) {
+      shopItemsView.setText(String.format(getString(R.string.item), itemCount));
+      return;
+    }
+    if (Objects.equals(groupName, GroupEnum.Travel.toString())) {
+      shopItemsView.setText(String.format(getString(R.string.item), itemCount));
+      return;
+    }
+    if (Objects.equals(groupName, GroupEnum.Bills.toString())) {
+      shopItemsView.setText(String.format(getString(R.string.item), itemCount));
+      return;
+    }
+    autoItemsView.setText(String.format(getString(R.string.item), itemCount));
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    updateAll(groups);
+    setupCurrentDate();
+    setupChart();
+    getFinishedPercentage();
+    setupDateFields();
+  }
+
+  private void getFinishedPercentage() {
+    taskController.getFinishPercentage(user, date).subscribe(new SingleSubscriber<Long>() {
+      @Override
+      public void onSuccess(Long percentage) {
+        setupCurrentPercentage(percentage.intValue());
+      }
+
+      @Override
+      public void onError(Throwable error) {
+        Timber.e(error);
+      }
+    });
+  }
+
+  private void setupCurrentPercentage(int percentage) {
+    //noinspection deprecation
+    SeriesItem seriesItem = new SeriesItem.Builder(getResources().getColor(R.color.seafoam_blue_two))
+        .setRange(0, 100, 0)
+        .setInterpolator(new DecelerateInterpolator())
+        .setSpinDuration(2000)
+        .setLineWidth(15f)
```

Comment:
The width is en float, then the duration is in milliseconds and the range is an int